src/controllers/queries/index.ts

```diff
@@ -321,33 +324,60 @@ export class QueriesController extends HTMLController {
     body: {
       companyId: UUID
       action: 'success'
-      totalScope3CarbonEmissions: string
+      emissions?: string
+      partialQuery?: 'on'[]
+      partialSelect?: 'on'[]
+      connectionIds?: string[]
+      productIds?: string[]
+      quantity?: string[]
     }
   ): Promise<HTML> {
-    req.log.info('query page requested %j', { queryId, body })
+    const { action, companyId, emissions, partialQuery, partialSelect, ...partial } = body
+    req.log.info('query page requested %j', { body })
 
-    const [connection] = await this.db.get('connection', { id: body.companyId, status: 'verified_both' }, [
+    const [connection]: ConnectionRow[] = await this.db.get('connection', { id: companyId, status: 'verified_both' }, [
       ['updated_at', 'desc'],
     ])
     if (!connection) {
-      req.log.warn('invalid input error %j', body)
-      throw new InvalidInputError(`Invalid connection ${body.companyId}`)
+      req.log.warn('invalid input error %j', { companyId, action })
+      throw new InvalidInputError(`Invalid connection ${companyId}`)
     }
     if (!connection.agent_connection_id || connection.status !== 'verified_both') {
-      req.log.warn('invalid input error %j', body)
+      req.log.warn('invalid input error %j', { companyId, action, emissions })
       throw new InvalidInputError(`Cannot query unverified connection`)
     }
-    const [queryRow] = await this.db.get('query', { id: queryId })
+    const [queryRow]: QueryRow[] = await this.db.get('query', { id: queryId })
     if (!queryRow.response_id) {
       req.log.warn('missing DRPC response_id to respond to %j', queryRow)
-      throw new InvalidInputError(`Missing queryId to respond to.`)
+      throw new InvalidInputError(`Missing response_id to respond to.`)
     }
 
-    const query = {
-      emissions: body.totalScope3CarbonEmissions,
+    const partialConnections: PartialQuery[] = []
+    if (partial.connectionIds && partial.productIds && partial.quantity && partialQuery) {
```

Comment:
I think this needs a bit of a logical change. If one is set but the others aren't that will be skipped which is odd. In that case I'd expect the request to error. Maybe consider setting default values for these body properties to `[]` and then simply always validating. Might make this simpler.

Second thing this new logic needs some new unit tests I think

Reply:
agreed, and tests have been included to assert different cases e.g. conncetionIds/quantities.